lib/tx-to-rest-converter.js

```diff
@@ -159,6 +161,83 @@ TxToRestConverter.prototype.parsePaymentFromTx = function(tx, options, callback)
   callback(null, payment);
 };
 
+/**
+ *  Convert an OfferCreate or OfferCancel transaction in rippled tx format
+ *  to a ripple-rest order_change
+ *
+ *  @param {Object} tx
+ *  @param {Object} options
+ *  @param {String} options.account - The account to use as perspective when parsing the transaction.
+ *
+ *  @callback
+ *  @param {Error} error
+ *  @param {Object} order_change - The parsed OrderChange transaction
+ *
+ *  @returns {Promise.<Object,Error>} - resolves to a parsed OrderChange transaction or an Error
+ */
+
+TxToRestConverter.prototype.parseOrderChangeFromTx = function(tx, options, callback) {
+  var promise = new Promise(function(resolve, reject) {
+    if (!options.account) {
+      reject(new Error('Internal Error. must supply options.account'));
+    }
+    if (tx.TransactionType !== 'OfferCreate' && tx.TransactionType !== 'OfferCancel') {
+      reject(new Error('Invalid parameter: identifier. The transaction corresponding to the given identifier is not an order'));
+    }
+    if (tx.meta !== void(0) && tx.meta.TransactionResult !== void(0)) {
+      if (tx.meta.TransactionResult === 'tejSecretInvalid') {
+        reject(new Error('Invalid secret provided.'));
+      }
+    }
+
+    var order;
+    var order_maker = tx.Account;
+    var flags = TxToRestConverter.prototype.parseFlagsFromResponse(tx.flags, OfferCreateFlags);
+    var order_type = tx.TransactionType === 'OfferCreate' ? 'order_create' : 'order_cancel';
+    var account_balance_changes = tx.meta ? parseBalanceChanges(tx.meta)[options.account] : [];
+    var order_maker_balance_changes = tx.meta ? parseBalanceChanges(tx.meta)[order_maker] : [];
+    var timestamp = tx.date ? new Date(ripple.utils.toTimestamp(tx.date)).toISOString() : '';
+    var order_book_changes = tx.meta ? parseOrderbookChanges(tx.meta) : [];
+
+    if (order_type === 'order_create') {
+      order = {
+        order_maker: order_maker,
+        taker_pays: utils.parseCurrencyAmount(tx.TakerPays),
+        taker_gets: utils.parseCurrencyAmount(tx.TakerGets),
+        passive: flags.passive,
+        immediate_or_cancel: flags.immediate_or_cancel,
+        fill_or_kill: flags.fill_or_kill,
+        type: flags.sell ? 'sell' : 'buy',
+        sequence: tx.Sequence,
+      };
+    } else {
+      order = {
+        order_maker: tx.Account,
+        type: 'cancel',
+        sequence: tx.Sequence,
+        cancel_sequence: tx.OfferSequence,
+      };
+    }
+
+    resolve({
+      account: options.account,
+      result: tx.meta ? tx.meta.TransactionResult : '',
+      hash: tx.hash,
+      ledger: tx.ledger_index,
+      validated: tx.validated,
+      timestamp: timestamp,
+      fee: utils.dropsToXrp(tx.Fee) || '',
+      type: order_type,
+      order: order,
+      account_balance_changes: account_balance_changes,
+      order_maker_balance_changes: order_maker_balance_changes,
+      order_book_changes: order_book_changes
+    });
+  });
+
+  return promise.nodeify(callback);
```

Comment:
`nodeify` is cool, not sure if we should have this ambiguous behavior though. I'd rather have one style of callback that we consistently apply.